### PR TITLE
Fix release notes build failure and PR preview

### PR DESCRIPTION
**Pull request by darrenchan**
Backport https://github.com/rcbops/rpc-openstack/pull/1365 to the mitaka-13.0 branch.

Connected https://github.com/rcbops/u-suk-dev/issues/384

### CODE REVIEW

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/staging.horse/build-0d664cf312/docs/private-cloud/rpc/release-notes/)

File: releasenotes/source/_deconst.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-    "contentIDBase": "https://github.com/rcbops/rpc-openstack/v13",
+    "contentIDBase": "https://github.com/rcbops/rpc-openstack/master/",
```
Comment (git-harry) [inline releasenotes/source/_deconst.json]: This branch is mitaka-13.0, does it matter that master is being referenced in multiple places?

**Comment by git-harry** [comment]
Why does the preview fail?

I'm not clear about what these changes are for. Is the purpose of the original patch and this backport to enable the creation of the preview? What are the ongoing maintenance requirement this introduces? The next time a new branch is created what steps need to be taken?

**Comment by rackernexus** [comment]
HULK [PREVIEW](https://staging.developer.rackspace.com/build-5b08580b2e/docs/private-cloud/release-notes/v13/)

**Comment by rackernexus** [comment]
:eyes: [Preview](https://staging.developer.rackspace.com/build-94659e3b0d/docs/private-cloud/release-notes/v13/)

**Comment by rackernexus** [comment]
If you want to double-check your formatting, check [the preview](https://staging.developer.rackspace.com/build-2fe41a3ceb/docs/private-cloud/release-notes/v13/).

**Comment by darrenchan** [comment]
@git-harry Preview failed because the contentIDBase did not match the content ID in https://github.com/rackerlabs/nexus-control/config to build previews. Yep, each branch should have a unique content ID.

The purpose of the previous patch and this backport/edit is to enable PR build preview, but also enable release notes on the branch to be published. For the master branch, it is published to a
draft staging site (https://staging.developer.rackspace.com/staging.horse/review/docs/private-cloud/release-notes/), based on the last merged commit. For mitaka-13.0, it will be published to developer.rackspace.com.

When a new branch is created:

- Edit _deconst.json with a unique _contentIDBase_ and update the _githubBranch_ variable
- Create a new release page and add it to the toctree in index.rst



You can inform myself, @meker12, @kallimachos, and @erikmwilson when a new branch has been created, and one of us can take care of  things.

Sorry for all the commits. I'm tweaking the formatting based on the generated previews.

**Comment by darrenchan** [comment]
@meker12 @smashwilson @odyssey4me If you look at the preview, I'm getting the template help text generated in each section. I removed the test note yaml file, but no luck. Any thoughts?

**Comment by darrenchan** [comment]
recheck

**Comment by git-harry** [comment]
@darrenchan we release new patch tags every two weeks. From your description it sounds like the tip of the branch will be included when publishing to developer.rackspace.com/docs/, this may include release notes that do not yet form part of a tag and therefore a release. Is this accounted for in the docs process?

The travis job is failing with:

```

[reno] scanning /home/travis/build/rcbops/rpc-openstack/releasenotes/notes for origin/mitaka-13.0 release notes
Exception occurred:
  File "/home/travis/build/rcbops/rpc-openstack/.tox/releasenotes/lib/python2.7/site-packages/reno/utils.py", line 60, in check_output
    raise subprocess.CalledProcessError(retcode, args, output=output)
CalledProcessError: Command '(['git', 'log', '--simplify-by-decoration', '--pretty="%d"', 'origin/mitaka-13.0'],)' returned non-zero exit status 128
The full traceback has been saved in /tmp/sphinx-err-rTEKCc.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
ERROR: InvocationError: '/home/travis/build/rcbops/rpc-openstack/.tox/releasenotes/bin/sphinx-build -a -E -W -d releasenotes/build/doctrees -b html releasenotes/source releasenotes/build/html'
___________________________________ summary ____________________________________
ERROR:   releasenotes: commands failed
```

I believe this is because travis is doing a shallow clone of master and so there are no refs for the other branches.

I think this might be fixed by adding the following in the before_script section of .travis.yml, although there may be better ways to solve the problem.
```
git fetch origin +refs/heads/*:refs/remotes/origin/*
```

Once I get past this error locally tox fails with:

```
Warning, treated as error:
/tmp/rcbops/rpc-openstack/releasenotes/source/unreleased.rst:: WARNING: document isn't included in any toctree

ERROR: InvocationError: '/tmp/rcbops/rpc-openstack/.tox/releasenotes/bin/sphinx-build -a -E -W -d releasenotes/build/doctrees -b html releasenotes/source releasenotes/build/html'
________________________________________________________________________________________________ summary _________________________________________________________________________________________________
ERROR:   releasenotes: commands failed
```

**Comment by rackernexus** [comment]
:mag_right: [Preview](https://staging.developer.rackspace.com/build-7f572e039f/docs/private-cloud/release-notes/v13/)

**Comment by git-harry** [comment]
@darrenchan please squash these down into one commit.

**Comment by smashwilson** [comment]
> @meker12 @smashwilson @odyssey4me If you look at the preview, the template help text is generated in each section. I removed the test note yaml file, but no luck. Any thoughts?

No idea, sorry - I've only used Reno just enough to get it running within a Deconst build.

**Comment by smashwilson** [comment]
Oh, actually, I bet it's coming from [this file](https://github.com/rcbops/rpc-openstack/blob/mitaka-13.0/releasenotes/notes/test-note-32a26392e1b9f585.yaml) on the mitaka-13.0 branch?

**Comment by meker12** [comment]
@darrenchan  To get rid of the error /tmp/rcbops/rpc-openstack/releasenotes/source/unreleased.rst:

You can put an :orphan: directive at the top of the file for a quick fix.

**Comment by git-harry** [comment]
@darrenchan what is happening with this? Do we still need to merge it if we are now doing things manually?

**Comment by darrenchan** [comment]
@git-harry We can still merge but there is no urgency now since we are creating the release notes manually for r13.0. Thanks, I'll squash my commits.

thanks @meker12 👍

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-6c667e0afe/docs/private-cloud/release-notes/v13/). :bow:

**Comment by rackernexus** [comment]
Your content preview [is now ready](https://staging.developer.rackspace.com/build-2f744a5f65/docs/private-cloud/release-notes/v13/). :bow:

**Comment by evrardjp** [comment]
tox'ing the env releasenotes works for me. no issues. correct building in this branch.

**Comment by git-harry** [comment]
👍 the preview appears to work correctly.

**Comment by meker12** [comment]
@darrenchan  The preview works, but it does still show the test data as you mentioned. 

Looking in the build results, I see this message:

[reno] unable to find file associated with unique id '32a26392e1b9f585', skipping
[reno] unable to find file associated with unique id '32a26392e1b9f585', skipping

That ID corresponds to the id on the test file: test-note-32a26392e1b9f585.yaml,  So I think there's something in Reno still looking for that ID?  

However, I don't think that's causing a problem with the build output: 

 If you check the preview on the master branch, the build log has the same message (only once), but the preview release notes don't have the boiler plate text. 

If you merge this PR, I am pretty sure the release note content from the production build will be OK.  Or submit a fresh PR with these files.  I've noticed that if you make nexus-control changes and then try to build from the same PR, the preview doesn't always contain what you expect.  @smashwilson  Might be able to help with this problem.

**Pull request merged**